pkg/sosreport/mcp/logs.go

```diff
@@ -0,0 +1,75 @@
+package sosreport
+
+import (
+	"os"
+	"strings"
+	"path/filepath"
+	"regexp"
+	"fmt"
+)
+
+// searchPodLogs searches pod logs using the manifest to find log files
+func searchPodLogs(sosreportPath, pattern, podFilter string, maxResults int) (string, error) {
+	searchPattern, err := regexp.Compile(pattern)
+	if err != nil {
+		return "", fmt.Errorf("invalid search pattern: %w", err)
+	}
+
+	manifest, err := loadManifest(sosreportPath)
+	if err != nil {
+		return "", err
+	}
+
+	if maxResults == 0 {
+		maxResults = 100
+	}
+
+	containerLogPlugin, exists := manifest.Components.Report.Plugins["container_log"]
+	if !exists || len(containerLogPlugin.Files) == 0 {
+		return "No pod logs found in sosreport\n", nil
+	}
+
+	var result strings.Builder
+	totalMatches := 0
+
+	for _, f := range containerLogPlugin.Files {
+		for _, logPath := range f.FilesCopied {
+			if podFilter != "" && !strings.Contains(logPath, podFilter) {
+				continue
+			}
+
+			// Remove the prefix if exists
+			logPath = strings.TrimPrefix(logPath, "host/")
+
+			fullPath := filepath.Join(sosreportPath, logPath)
```

Comment:
we may also have rotated log files in .gz format, shouldn't that taken care of ?